datafusion/core/src/datasource/physical_plan/parquet/statistics.rs

```diff
@@ -713,6 +692,15 @@ macro_rules! get_data_page_statistics {
                 )),
                 Some(DataType::Float32) => Ok(Arc::new(Float32Array::from_iter([<$stat_type_prefix Float32DataPageStatsIterator>]::new($iterator).flatten()))),
                 Some(DataType::Float64) => Ok(Arc::new(Float64Array::from_iter([<$stat_type_prefix Float64DataPageStatsIterator>]::new($iterator).flatten()))),
```

Comment:
Should this repeated code be extracted? I assume it would need to be put into a macro?

Reply:
Which repeated code do you mean? 

If you wnated to more fully macroize this table and avoid repetition that sounds like a great idea to me (though perhaps as a follow on PR)

Reply:
The four lines in the match unit expression are exactly the same in both macros.